### DISCUSSION TITLE
Unexplainable cacher-behaviour

### DISCUSSION THREAD

**Discussion by eawmaas** [xenocrat/chyrp-lite#201] [Q&A] [answered]
I have had some weird behaviour with caching the last few weeks but somehow they went away pretty quickly (like within 15 minutes).

But yesterday-evening it was a stubborn one: I made some changes to the production-branche (later more about that in another request / discussion) and Chrome kept on serving the old version, also after I emptied browser-cache.

I disabled cacher and tried again, still the same old version.

I let the cacher-module OFF and tried with Edge and Firefox: same old version again.

Then I deleted all the cache-maps in includes/caches/twig and that worked, the changes I made were finally visible in the source-code.

But: before shutting down yesterday evening there were around 8 new directories in includes/caches/twig with the cacher-module still disabled and this morning there are 16 allready.

Will it be a good idea if I try to make a mod or make changes to the cacher-mod to flush the whole cache, meaning delete all these maps instantly with one push on a button?

**Comment by xenocrat** [comment, marked as answer]
Hello there,

The contents of _inlcudes/caches/twig_ are created by the Twig template engine used for rendering themes. They don't have anything to do with the cacher module. If you are making a lot of changes or developing a theme, you'll find it convenient to enable DEBUG mode in Chyrp Lite. DEBUG mode will cause Twig to recheck all template files for changes on each request instead of relying on its caches.

The cacher module is actually very simple and doesn't do any caching at all on the server; instead it tells the web browsers of your visitors to keep a local cached copy of the pages it sends to them. If your visitors return to the same page and nothing has changed on the site, then the module tells the browsers to continue using their cached copies. This avoids redoing all the computation and database access needed to respond to a request.

**Reply by eawmaas** [reply]
That's clear enough, now I understand another unexplained thing too (some changes did come through with DEBUG-mode on but not in normal-mode...)

Thanks for the explanation!
